.github/workflows/build_test_push.yaml

```
@@ -88,3 +146,17 @@ jobs:
 
       - name: run functional test
         run:  bash functional_test.sh ${{ github.sha }}-server-dev
+  add-git-tags:
+    runs-on: ubuntu-latest
+    needs: functional-tests
+    steps:
+    - name: Bump version and push tag
+      id: create_tag_step
+      uses: anothrNick/github-tag-action@1.26.0
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        WITH_V: true
+        INITIAL_VERSION: 1.0.0
+        RELEASE_BRANCHES: original-direction
+    - name: "Use tag. example: could reuse git tag as docker image tag"
+      run: echo ">> ${{ steps.create_tag_step.outputs.new_tag }}"
```

CalVer alternative w/o an action:
```suggestion
  add-git-tags:
    if: ${{ github.ref == 'refs/heads/original-direction' }}
    needs: functional-tests
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Bump version and push tag
        run: |
          git config user.name "Replica Robots"
          git config user.email "robots@replicahq.com"
          tag="$(date -u '+%Y-%m-%dT%H-%M')"
          git tag "${tag}"
          git push origin "${tag}"
          echo ">> "${tag}""
```
Benefits:
- calver is explicit about the lack of true meaning (rather than a "pseudo semantic" semver)
- works ok w/ frequent merged (ok, pretty infrequent in this repo) w/o "inflation" like `v0.104.0`

---

OK, cool, sounds like you're not a fan of semver specifically, both out of aesthetic preferences and some concerns about hypothetical future workflow problems. That's about how I feel about using dates as versions. But, I think we're in agreement that we don't really _need_ the major/minor/patch features of semver for this use-case, and likely won't want them at all for versioning the model repo. So, here's a suggestion - how about I take the time to implement an incrementing integer version? That's a scheme that I would be happy with, and would remove a lot of the semver-specific stuff you were concerned about. How does that sound?

---

If you're not convinced by date (or `date -u '+%s'`->`1627667261`), then yeah that sounds like a good alternative if it's not too much work. One simple thing might be to replace the `tag` var in my suggestion above with `tag="${GITHUB_RUN_NUMBER}"`:
> A unique number for each run of a particular workflow in a repository. This number begins at 1 for the workflow's first run, and increments with each new run. This number does not change if you re-run the workflow run.

Admittedly, I'm a bit of "strong opinion weakly held" here - wouldn't love violating semver promises, but won't be too bothered if we went with that. 😄)